tests/dashbio_demos/app_ideogram.py

```diff
@@ -14,833 +15,750 @@
-            Input("chrone-stoptwo", "value"),
-            Input("chrtwo-startone", "value"),
-            Input("chrtwo-stopone", "value"),
-            Input("chrtwo-starttwo", "value"),
-            Input("chrtwo-stoptwo", "value"),
-        ],
-    )
-    def ideo_homology(
-            start_one,
-            stop_one,
-            start_two,
-            stop_two,
-            start_one_a,
-            stop_one_a,
-            start_two_a,
-            stop_two_a
-    ):
-        return {
-            "chrOne": {
-                "organism": "9606",
-                "start": [start_one, start_two],
-                "stop": [stop_one, stop_two],
-            },
-            "chrTwo": {
-                "organism": "9606",
-                "start": [start_one_a, start_two_a],
-                "stop": [stop_one_a, stop_two_a],
-            },
-        }
-
-    # Brush Callbacks
-    @app.callback(
-        Output("brush-ideo", "chromosomes"),
-        [Input("chr-brush", "value")]
-    )
-    def ideo_select_chr(value):
-        answer = ["1"]
-        if value:
-            value = str(value)
-            answer = [value]
+    def brush_data_start(brush_data):
+        answer = None
+        if brush_data is not None:
+            answer = brush_data['start']
         return answer
 
     @app.callback(
-        Output("brush-ideo", "brush"),
-        [Input("chr-brush", "value")]
+        Output('brush-print-end', 'children'),
+        [Input('brush-ideo', 'brushData')]
     )
-    def ideo_select_brush(value):
-        answer = "chr1:1-10000000"
-        if value:
-            value = "chr{}:1-10000000".format(value)
-            answer = value
+    def brush_data_end(brush_data):
+        answer = None
+        if brush_data is not None:
+            answer = brush_data['end']
         return answer
 
-    # Event callbacks
-    # Color Change onRotate
     @app.callback(
-        Output("ideogram-options", "style"),
-        [Input("ideo-custom", "rotated")],
-        [State("ideogram-options", "style")],
+        Output('brush-print-extent', 'children'),
+        [Input('brush-ideo', 'brushData')]
     )
-    def ideo_rotated(value, _):
-        if value:
-            style = {"color": "#EF553B"}
-        else:
-            style = {"color": "#000"}
-        return style
+    def brush_data_extent(brush_data):
+        answer = None
+        if brush_data is not None:
+            answer = brush_data['extent']
+        return answer
 
     # Event Call Annotation
     @app.callback(
-        Output("annote-data", "children"),
-        [Input("ideo-annotations", "annotationsData")],
+        Output('annote-data', 'children'),
+        [Input('ideo-annotations', 'annotationsData')],
     )
     def annote_data(data):
         if data is None:
-            data = "None"
-        elif "<br>" in data:
-            data = data.split("<br>")
-            data = data[0] + " " + data[1]
+            data = 'None'
+        elif '<br>' in data:
```

Comment:
When `'<br>' in data`, is it always certain that there's only one instance of `'<br>'`?

Reply:
As far as I can tell, yes, but making the code more general will be a quick fix :)

Reply:
Fixed here: 2e163d9